meson.build

```diff
@@ -0,0 +1,82 @@
+project(
+    'dwave-optimization',
+    'cpp', 'cython',
+    default_options: [
+        'cpp_std=c++20',
+        'buildtype=release',
+        'b_ndebug=if-release',  # add -DNDEBUG when building for "release", which we do by default
+        'pkgconfig.relocatable=true',
+    ],
+
+    # As of August 2024, there is no nicer way that I know of to dynamically get the version
+    version: run_command(
+        ['dwave/optimization/_build/_version.py'],
+        capture: true,
+        check: true,
+    ).stdout(),
+)
+
+# We want some debugging symbols
+add_project_arguments(['-g1'], language: ['cpp'])
+
+py = import('python').find_installation(pure: false)
+
+dwave_optimization_include = include_directories('dwave/optimization/include/')
+dwave_optimization_src = [
```

Comment:
To head off an obvious question, [meson doesn't allow wildcards](https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard).